### PR TITLE
SDK-45: Java sdk: Add support for ClusterManage Command

### PR DESCRIPTION
**Pull request by abhisheksr8**

### CODE REVIEW

**Comment by amoghmargoor** [comment]
Please add examples for this as customers were asking for it in different instance.

**Comment by abhisheksr8** [comment]
added examples and tests

File: src/main/java/com/qubole/qds/sdk/java/details/ClusterApiImpl.java [hunk 1]
```diff
@@ -83,6 +88,28 @@
     }
 
     @Override
+    public InvokableBuilder<Command> add_nodes(String labelOrId, int node_count)
+    {
+        RequestDetails entity = new RequestDetails(new AddNode(node_count), RequestDetails.Method.POST);
+        System.out.println(entity.getEntity());
```
Comment (amoghmargoor) [inline src/main/java/com/qubole/qds/sdk/java/details/ClusterApiImpl.java]: No println in the code (apart from tests or examples)

File: src/main/java/com/qubole/qds/sdk/java/details/ClusterApiImpl.java [hunk 2]
```diff
@@ -83,6 +88,28 @@
...
+        return new GenericInvokableBuilderImpl<Command>(client, entity, Command.class, "clusters", labelOrId, "nodes");
+    }
+
+    @Override
```
Comment (amoghmargoor) [inline src/main/java/com/qubole/qds/sdk/java/details/ClusterApiImpl.java]: Doc for all 3 are API are missing. Its self explanatory, but would not look good in generated javadoc to have empty description.

File: src/main/java/com/qubole/qds/sdk/java/entities/AddNode.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright 2014- Qubole Inc.
```
Comment (amoghmargoor) [inline src/main/java/com/qubole/qds/sdk/java/entities/AddNode.java]: Whats the norm followed for copyright thingy ? 2014- current year ?

File: src/test/java/com/qubole/qds/sdk/java/unittests/TestCluster.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        String randomclusterId="123";
+        String dnsOfNode="ec2-123-123-123-123.qubole.com";
+        InvokeArguments<Command> invokeargs = qdsClient.cluster().replace_node(randomclusterId, dnsOfNode).getArgumentsInvocation();
+        JSONObject params=new JSONObject();
+        params.put("private_dns", dnsOfNode);
+        params.put("command", "replace");
+        assertRequestDetails(invokeargs, "PUT", "clusters/"+randomclusterId+"/nodes", params, null, Command.class);
+    }
+
+    public void assertRequestDetails(InvokeArguments<?> invokeargs, String expectedRequestType, String expectedEndpoint, JSONObject expectedRequestData, Map<Object, Object> expectedQueryParams, Class<?> expectedResponseClassType) throws Exception
```
Comment (amoghmargoor) [inline src/test/java/com/qubole/qds/sdk/java/unittests/TestCluster.java]: Looks like this is a common function. I think we should create a abstract class having this method which will be inherited by rest of the Test classes. We can put common checking code there in abstract class.

File: src/main/java/com/qubole/qds/sdk/java/details/ClusterApiImpl.java (hunk 2 above)
Comment (abhisheksr8) [inline src/main/java/com/qubole/qds/sdk/java/details/ClusterApiImpl.java]: docs for all 3 apis are in Clusterapi class

File: src/main/java/com/qubole/qds/sdk/java/details/ClusterApiImpl.java (hunk 1 above)
Comment (abhisheksr8) [inline src/main/java/com/qubole/qds/sdk/java/details/ClusterApiImpl.java]: added for test purpose, forgot to remove,  will remove

File: src/main/java/com/qubole/qds/sdk/java/entities/AddNode.java (hunk 3 above)
Comment (abhisheksr8) [inline src/main/java/com/qubole/qds/sdk/java/entities/AddNode.java]: No its jusr Copyright 2014-<name of company i.e Qubole Inc>

File: src/test/java/com/qubole/qds/sdk/java/unittests/TestCluster.java (hunk 4 above)
Comment (abhisheksr8) [inline src/test/java/com/qubole/qds/sdk/java/unittests/TestCluster.java]: made changes ad part of SDK-48

**Comment by amoghmargoor** [comment]
There are 8 test failures. Can you check what happened ?

**Comment by abhisheksr8** [comment]
it is parsing issue with jackson, i have made some workarounds to get around the issue

**Pull request merged**